src/Dofs/ConstraintHandler.jl

```diff
@@ -40,7 +40,33 @@ end
 function Dirichlet(field_name::Symbol, faces::Set{Tuple{Int,Int}}, f::Function, components::Vector{Int})
     unique(components) == components || error("components not unique: $components")
     # issorted(components) || error("components not sorted: $components")
-    return Dirichlet(f, faces, field_name, components, Int[], Int[])
+    return FaceDirichlet(f, faces, field_name, components, Int[], Int[])
+end
+
+"""
+    NodeDirichlet
+
+A Dirichlet boundary condition is a boundary where the solution is fixed to take a certain value.
+
+`NodeDirichlet` boundary conditions are added to certain components of a field for a specific nodes of the grid.
+A function is also given that should be of the form `(x,t) -> v` where `x` is the coordinate of the node, `t` is a
+time parameter and `v` should be of the same length as the number of components the bc is applied to.
+
+"""
+struct NodeDirichlet
```

Comment:
Instead of storing the nodes in `Set{Int}`, one could maybe store it in a similar ways as the faces: ``nodes::Set{Tuple{ cellid , local_node_number}}`. Then I think the dofhandler does not need to store what nodes has which dofs, and NodeDirichlet and FaceDirichlet can be combined

Reply:
The 2 structs are given below, apart from nodes and faces whose type can be made a type parameter, how do you suggest dealing with the other mismatching fields.

```julia
struct FaceDirichlet # <: Constraint
    f::Function # f(x,t) -> value
    faces::Set{Tuple{Int,Int}}
    field_name::Symbol
    components::Vector{Int} # components of the field
    local_face_dofs::Vector{Int}
    local_face_dofs_offset::Vector{Int}
end
struct NodeDirichlet
    f::Function
    nodes::Vector{Int}
    field_name::Symbol
    components::Vector{Int}
    dofs::Vector{Int}
end
```

Reply:
Regardless of this, I think it is useful to have `node_dofs` stored for easy of post-processing. Since it should be possible to query the `dofs` of a node without going through a search for a cell that contains that node.

Reply:
One more comment here is that (the cell_id, local_node_id) is not a unique identifier for a boundary node, unlike for faces, since a boundary face can only be part of one cell by definition. And node BCs in .inp files for example only give the node number, so finding the cell requires a linear time search even when adding a single node, since one may have to scan all cells in the worst case looking for the (cell_id, local_node_id) non-unique identifier to use instead. The current way of adding node BCs is of constant time complexity for a single node.